app/components/Authentication/Github/Github.jsx

```diff
@@ -1,248 +1,513 @@
+                    .then(() => {
+                        snackBarMessage(`GitHub ${this.state.selectedTab} ${id} has been updated`);
+                        this.listGithubTeams();
+                        this.listGithubUsers();
+                        this.setState({ openItemDialog: false, openNewItemDialog: false, itemConfig: _.clone(this.itemConfigSchema), selectedItemId: '' });
+                        browserHistory.push(this.state.baseUrl);
+                    })
+                    .catch(snackBarMessage);
+            })
+            .catch(() => {
+                this.setState({ selectedRoleId: '' })
+                snackBarMessage(new Error(`No permissions to display properties for role ${id}`));
             })
-            .catch((err) => {
-                this.setState({ errorMessage: `${err} - URI: ${decodeURI(uri)}` });
-            });
     }
 
-    teamNameChanged(e, v) {
-        if (v === '') {
-            _.forEach(this.state.policies, (policy) => {
-                policy.checked = false;
-            });
-
-            this.setState({
-                policies: this.state.policies
-            });
+    componentDidMount() {
+        if (this.props.params.splat) {
+            this.setState({ selectedItemId: this.props.params.splat });
         } else {
-            this.state.teamName = v;
-            this.processTeamNameDebounced();
+            this.listGithubTeams();
+            this.listGithubUsers();
         }
+        this.getOrgConfig();
+    }
 
+    componentDidUpdate(prevProps, prevState) {
+        if (this.state.selectedItemId != prevState.selectedItemId) {
+            this.listGithubTeams();
+            this.listGithubUsers();
+            if (this.state.selectedItemId) {
+                this.displayItem();
+            }
+        }
     }
 
-    processTeamName() {
-        this.getTeamPolicy(this.state.teamName);
-        this.setState({
-            teamName: this.state.teamName,
-            submitBtnDisabled: !(this.state.checkedPolicies && this.state.organization && this.state.teamName),
-            submitBtnColor: (this.state.checkedPolicies && this.state.organization && this.state.teamName) ? green500 : 'lightgrey'
-        });
+    componentWillReceiveProps(nextProps) {
+        if (!_.isEqual(this.props.params.namespace, nextProps.params.namespace)) {
+            // Reset
+            this.setState({
+                baseUrl: `/auth/github/${nextProps.params.namespace}/`,
+                baseVaultPath: `auth/${nextProps.params.namespace}`,
+                users: [],
+                teams: [],
+                selectedItemId: '',
+                newConfig: this.backendConfigSchema,
+                config: this.backendConfigSchema,
+                selectedTab: "team"
+            }, () => {
+                this.listGithubTeams();
+                this.listGithubUsers();
+                this.getOrgConfig();
+            });
+        }
```

Comment:
It allows switching between two of the same auth backends mounted at different paths